app/controllers/import_wizards_controller.rb

```diff
@@ -81,4 +88,43 @@ def job_status
   def logs
     add_breadcrumb "Import wizard", collection_import_wizard_path(collection)
   end
+
+  private
+
+  def sheet_id_match(url)
+    match = url.match /^.*\/d\/(?<sheetId>.*)\/.*$/
+    match && match[:sheetId]
+  end
+
+  def import_csv_from_file(filename, file_content)
+    begin
+      csv = ImportWizard.import current_user, collection, filename, file_content
+      if csv.length == 1
+        message = "The uploaded csv is empty."
+        redirect_to adjustments_collection_import_wizard_path(collection), :notice => message
+      else
+        redirect_to adjustments_collection_import_wizard_path(collection)
+      end
+    rescue => ex
+      redirect_to collection_import_wizard_path(collection), :alert => ex.message
+    end
+  end
+
+  def from_google_spreadsheet(spread_sheet_link)
+    filename = "#{current_user.id}_#{collection.id}.csv"
```

Comment:
Does this work if the same user imports a spreadsheet into a collection after having imported a previous one? What potential conflicts could arise then?

Reply:
The file's content gets overwritten with the spreadsheet content that the user inputs. Filename does not change, but content does. We could use the spreadsheet id as the filename, but then we will have stored a csv file for every single spreadsheet users want to import. 
With the current code we will have one spreadsheet per user that used the feature as the worst case scenario.